Prompt Windows UAC when updating in a privileged folder.

cameronmaske opened the pull request:
Hey there, 
First off, thanks so much for PyUpdater, it's a much-needed package in the python ecosystem and has been immensely useful in my own projects! 

This patch should address an issue in windows, when the user has installed the `.exe` in a privileged folder (e.g. `C://Program Files` aka #35) and the update fails silently. 
It should prompt the user for access. 

I need to do some testing to verify this works on my end and will update this PR with functional proof.

cameronmaske committed:
On windows, when updating, detect if admin permissions are required and if so, run the script with elevated privileged.

cameronmaske committed:
When update is folder, check admin

cameronmaske committed:
Removed debugging logging

cameronmaske committed:
Experimental test

cameronmaske committed:
Change file_require_admins to work more reliably with windows

cameronmaske committed:
Switch from os.popen4 to python2/3 compatible approach

2press committed:
Adding compatibility with Pyton 3

cameronmaske committed:
Merge pull request #1 from 2press/admin-update-2

Adding compatibility with Python 3

cameronmaske committed:
Check if can append file

cameronmaske commented on pyupdater/client/updates.py:
```diff
@@ -49,17 +49,12 @@
-    )
-    output = process.stdout.read().decode("utf-8")
-    return "Access is denied" in output
+    try:
```
Made a change passed on the comment from https://github.com/JMSwag/PyUpdater/pull/116#issuecomment-384422747
Need to test if this operates in the same way....

2press replied:
Works for me on Python 2 and 3 without special chars and on Python 3 with special chars, but not on Python 2 with special chairs. But making the adjustment `with open(file_path.decode('utf-8'), "a"):`  to this, works for me on Python 2 with special chars as well.

cameronmaske replied:
I'll make that change. How are you testing this?

2press replied:
I am testing this by copying the `file_require_admin` method into a new small python script and calling it for local files on my system.

cameronmaske committed:
decode utf-8 on file path

cameronmaske committed:
Add verbose logging to pytest@

cameronmaske committed:
Add print out to pytest

cameronmaske committed:
Handle new permission error handling in Python 3 due to PEP 3151

cameronmaske committed:
More debugging

cameronmaske committed:
Add even more debugging

cameronmaske committed:
Check if python 2 passes

cameronmaske committed:
Merge branch 'master' into admin-update-2

cameronmaske commented on tests/test_pyupdater.py:
```diff
@@ -44,6 +44,8 @@
 LOCK_TIMEOUT = 5*60  # ten minutes timeout
 APP_NAME = 'Acme'
 
+import logging
```
TODO remove this

cameronmaske commented on tox.ini:
```diff
@@ -33,7 +33,11 @@ recreate = True
 addopts =
+    -x
     -n 4
+    -s
```
TODO remove these changes

cameronmaske committed:
Update tox.ini

cameronmaske committed:
Update test_pyupdater.py

2press committed:
Update requirements.txt

2press committed:
Update requirements.txt

2press committed:
Update __init__.py

2press committed:
Update keys.py

2press committed:
Update requirements.txt

2press committed:
Update requirements.txt

2press committed:
Update __init__.py

2press committed:
Update test_version_file.py

2press committed:
Update requirements.txt

2press committed:
Use pure25519 instead of ed25519 only on Windows 64-bit.

2press committed:
Fixed setup.py

2press committed:
Update setup.py

2press committed:
Update settings.py

2press committed:
Update test_version_file.py

2press committed:
Update test_version_file.py

2press committed:
Update setup.py

2press committed:
Update test_version_file.py

2press committed:
Update settings.py

2press committed:
Update .travis.yml

2press committed:
Ensure appveyor uses correct version of pip/pytest

`python -m pip` seems to be using a different version of pip as the newer version that pip has been updated to is not used.

2press committed:
appveyor: added --ignore-installed to PyUpdater

2press committed:
Fixed check for 64bit

cameronmaske committed:
Merge pull request #2 from 2press/patch-1

Update test/requirements.txt for travis

cameronmaske committed:
Update .travis.yml

2press committed:
Try #1 to set up AppVeyor correctly

2press committed:
Fixed parsing error for appveyor.yml

2press committed:
Fix appvevyor.yml

2press committed:
Try #2 to set up AppVeyor correctly

2press committed:
Readded TOXENV to appveyor.yml

2press committed:
Merge branch 'admin-update-2' of https://github.com/cameronmaske/PyUpdater into cameronmaske-admin-update-2

2press committed:
Merge branch 'cameronmaske-admin-update-2' into my-admin-fix

2press committed:
Fixed merge

2press committed:
Added whitespaces

2press committed:
Removed decode for python3

2press committed:
Added pywin32 to requirements for Windows

2press committed:
Automatic distinction between file and dir for admin rights.

2press committed:
An executable that is currently being run gives a false positive for file_requires_admin. Therefore check it's directory instead.

cameronmaske committed:
Merge pull request #3 from 2press/my-admin-fix

Was able to fix some issues

2press committed:
Workaround for pip 19 issue

See https://github.com/pypa/pip/issues/6163

cameronmaske committed:
Merge pull request #4 from 2press/patch-1

Workaround for pip 19 issue

2press committed:
Update setup.py

2press committed:
Added pywin32 (only on Windows) to requirement.txt

2press committed:
Update requirements.txt

2press committed:
Merge pull request #3 from cameronmaske/admin-update-2

Admin update 2

2press committed:
Update requirements.txt

2press committed:
Pin pip to 18.1 for appveyor

2press committed:
Pinned pip to 18.1 for travis

2press committed:
Reverted pip pin

2press committed:
Reverted pip pin for appveyor

cameronmaske committed:
Merge pull request #5 from 2press/my-admin-fix

Pinned pip to 18.1 & refactored pywin32 requirement
